ocpp/charge_point.py

```diff
@@ -69,6 +69,85 @@ def snake_to_camel_case(data):
     return data
 
 
+def _is_dataclass_instance(input: Any) -> bool:
+    """Verify if given `input` is a dataclass."""
+    return is_dataclass(input) and not isinstance(input, type)
+
+
+def _is_optional_field(field: Field) -> bool:
+    """Verify if given `field` allows `None` as value.
+
+    The fields `schema` and `host` on the following class would return `False`.
+    While the fields `post` and `query` return `True`.
+
+        @dataclass
+        class URL:
+            schema: str,
+            host: str,
+            post: Optional[str],
+            query: Union[None, str]
+
+    """
+    return get_origin(field.type) is Union and type(None) in get_args(field.type)
+
+
+def serialize_as_dict(dataclass, remove_empty_optional_fields: bool = True):
```

Comment:
I think the solution can be simplified.

The root of issue #255 is that `dataclasses.asdict()` only serializes the top level dataclass as a dictionary. Any inner dataclasses are not serializes as a dict. 

The body of this function fixes that: recursively serializing a dataclasses as a dictionary.

But this function does more:
1. remove any optional fields containing `None`.
2. change casing to snake_case.

I think we can remove the latter to features of this function.

And, in lines [318](https://github.com/mobilityhouse/ocpp/blob/71f57ea18a748743aaf1ec185525f17fddd2fc2b/ocpp/charge_point.py#L318) and [378](https://github.com/mobilityhouse/ocpp/blob/71f57ea18a748743aaf1ec185525f17fddd2fc2b/ocpp/charge_point.py#L373),  replace the call to `dataclass.asdict()` with `serialize_as_dict()`.

Reply:
asdict() corrected in commit d1609599424af910fd596081afaafdc8fcef546d

Reply:
Removed the more with commit 2834dbefd92f9e072df5cdf81a163b96daef8a07